### PR TITLE
wip camera editor

### PR DESCRIPTION
**Pull request by paulsapps**

### CODE REVIEW

File: Source/editortab.cpp [hunk 1]
```diff
@@ -240,6 +248,24 @@ class EditorGraphicsView : public QGraphicsView
...
+
+    void contextMenuEvent(QContextMenuEvent* pEvent) override
+    {
+        QMenu menu(this);
+        auto pAction = new QAction("Edit camera", &menu);
+        connect(pAction, &QAction::triggered, this, [&]()
+            {
+                const QPoint scenePos = mapToScene(pEvent->pos()).toPoint();
+                qDebug() << "scenePos = " << scenePos;
+                CameraManager camerageManager(this, mEditorTab, &scenePos);
```
Comment (mouzedrift) [inline Source/editortab.cpp:260]: ``camerageManager`` typo 👀

File: Source/cameramanager.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        return mCamera;
+    }
+private:
+    Camera* mCamera = nullptr;
+};
+
+CameraManager::CameraManager(QWidget *parent, EditorTab* pParentTab, const QPoint* openedPos) :
+    QDialog(parent),
+    ui(new Ui::CameraManager),
```
Comment (d3nwah) [inline Source/cameramanager.cpp:27]: Wouldn't it be more convenient to smart_ptr it?

File: Source/editortab.cpp (hunk 1 above)
Comment (d3nwah) [reply Source/editortab.cpp:260]: camerage carnage!

File: Source/cameramanager.cpp (hunk 2 above)
Comment (paulsapps) [reply Source/cameramanager.cpp:76]: its generated code and blows up QtCreator if you change it

**Pull request merged**